### ISSUE TITLE
WPDB Error "Unknown column 'Datetime.RCR_ID' in 'field list'" while running wpdb method "get_results"

### ISSUE THREAD

**Issue by Pebblo** [eventespresso/eea-recurring-events-manager#48]
Latest dev version of core and rem.

If I view the event list (Event Espresso -> Events), I'm getting this notice:

`Notice: WPDB Error "Unknown column 'Datetime.RCR_ID' in 'field list'" while running wpdb method "get_results" with arguments ["SELECT Datetime.DTT_ID AS 'Datetime.DTT_ID', Datetime.EVT_ID AS 'Datetime.EVT_ID', Datetime.DTT_name AS 'Datetime.DTT_name', Datetime.DTT_description AS 'Datetime.DTT_description', Datetime.DTT_EVT_start AS 'Datetime.DTT_EVT_start', Datetime.DTT_EVT_end AS 'Datetime.DTT_EVT_end', Datetime.DTT_reg_limit AS 'Datetime.DTT_reg_limit', Datetime.DTT_sold AS 'Datetime.DTT_sold', Datetime.DTT_reserved AS 'Datetime.DTT_reserved', Datetime.DTT_is_primary AS 'Datetime.DTT_is_primary', Datetime.DTT_order AS 'Datetime.DTT_order', Datetime.DTT_parent AS 'Datetime.DTT_parent', Datetime.DTT_deleted AS 'Datetime.DTT_deleted', Datetime.RCR_ID AS 'Datetime.RCR_ID', Event_CPT.ID AS 'Event_CPT.ID', Event_CPT.post_title AS 'Event_CPT.post_title', Event_CPT.post_content AS 'Event_CPT.post_content', Event_CPT.post_name AS 'Event_CPT.post_name', Event_CPT.post_date AS 'Event_CPT.post_date', Event_CPT.post_excerpt AS in C:\laragon\www\ee4editor\wp-content\plugins\event-espresso-core\core\db_models\EEM_Base.model.php on line 2394`

The event this is happening on was created before REM was activated.

The DB table does have the `esp_recurrence` table and correct columns.

**Comment by tn3rb** [comment]
![have you tried turning it off and then on again](https://user-images.githubusercontent.com/1751030/113334561-b314ad00-92d8-11eb-94f1-90ea6426afd4.jpg)

ie: what about after de-re-activating EE ?

**Comment by Pebblo** [comment]
Nope, I haven't changed branches, confirmed this on my main EDTR site and a brand new DB (same site just changed the DB table and started clean) site is (and always has) using dev.

De-re-activating EE and/or REM does nothing.

If the `esp_recurrence` table has been created doesn't that means the activation script did run?

**Comment by Pebblo** [comment]
Ok, I've found out why. 

`INFORMATION_SCHEMA.COLUMNS` is across all databases.

REM's DMS script uses this:

`$row = $wpdb->get_results(
    "SELECT COLUMN_NAME
        FROM INFORMATION_SCHEMA.COLUMNS 
        WHERE TABLE_NAME = '{$table}'
        AND COLUMN_NAME = 'RCR_ID';"
);`

Which at first looks fine, but if I echo output `$row`, it does return a row even though `esp_datetime` on this site does NOT have the `RCP_ID` column.

Switching it out to be this:

`$row = $wpdb->get_results(
    "SELECT *
        FROM INFORMATION_SCHEMA.COLUMNS 
        WHERE TABLE_NAME = '{$table}'
        AND COLUMN_NAME = 'RCR_ID';"
);`

Obviously shows me the full row returned and this is where things get interesting.

Heres the output:

`public TABLE_CATALOG -> string (3) "def"
public TABLE_SCHEMA -> string (11) "wpmultisite"
public TABLE_NAME -> string (15) "wp_esp_datetime"
public COLUMN_NAME -> string (6) "RCR_ID"
public ORDINAL_POSITION -> string (2) "14"
public COLUMN_DEFAULT -> NULL
public IS_NULLABLE -> string (3) "YES"
public DATA_TYPE -> string (3) "int"
public CHARACTER_MAXIMUM_LENGTH -> NULL
public CHARACTER_OCTET_LENGTH -> NULL
public NUMERIC_PRECISION -> string (2) "10"
public NUMERIC_SCALE -> string (1) "0"
public DATETIME_PRECISION -> NULL
public CHARACTER_SET_NAME -> NULL
public COLLATION_NAME -> NULL
public COLUMN_TYPE -> string (16) "int(10) unsigned"
public COLUMN_KEY -> string (3) "MUL"
public EXTRA -> string (0) ""
public PRIVILEGES -> string (31) "select,insert,update,references"
public COLUMN_COMMENT -> string (0) ""
public GENERATION_EXPRESSION -> string (0) ""`

The interesting part is `public TABLE_SCHEMA -> string (11) "wpmultisite"`

The DB name for the site I'm testing this on is `ee4editorlatest` and `wpmultisite` is the site I set up to check the DMS issue the other day.

**Comment by tn3rb** [comment]
can you plz provide a detailed list of steps to reproduce the issue?

I tried the following:

- created a branch new site
- activated EE core
- activated REM
- activated Query Monitor (for help with viewing all db errors and warnings)
- visited event list
- checked QM for errors (none)
- created a standard non-REM event and saved it
- returned to event list
- checked QM for errors (none)
- edited previously created event, added a recurring date series, saved it
- returned to event list
- checked QM for errors (none)

Is this just on your local test site, or are you also seeing this error on a web accessible site that we can troubleshoot?

**Comment by tn3rb** [comment]
somehow missed the last sentence of your last post...
I'll try adding the site name into the query and see if that helps

**Comment by tn3rb** [comment]
@Pebblo 

Can you plz chance the one query in `\EE_DMS_Recurring_Events_1_0_0::schema_changes_before_migration()` to:

```php
"SELECT COLUMN_NAME
FROM INFORMATION_SCHEMA.COLUMNS 
WHERE TABLE_SCHEMA = '{$wpdb->dbname}'
AND TABLE_NAME = '{$table}'
AND COLUMN_NAME = 'RCR_ID';"
```

and see if that works plz and thnx?

**Comment by sethshoultes** [comment]
@Pebblo @tn3rb any luck with figuring out this issue?

**Comment by Pebblo** [comment]
#49 fixes this for me.

**Comment by Pebblo** [comment]
Merged #49 so closing this out.
